### PR TITLE
Postmigration script cleanup 7.17

### PR DESCRIPTION
**Pull request by oakrizan**
## Proposed commit message
Removed unused/unnecessary scripts and updated its' usage in other corresponding scripts.
`deploy/kubernetes` test failure is not related to changes in current PR (https://github.com/elastic/ingest-dev/issues/3549)

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates: https://github.com/elastic/ingest-dev/issues/3219

## Logs
BK builds: https://buildkite.com/elastic/beats/builds?branch=oakrizan%3Apostmigration-script-cleanup

### CODE REVIEW

File: .buildkite/hooks/pre-exit [hunk 1]
```diff
@@ -2,14 +2,23 @@
 
 set -euo pipefail
 
-source .buildkite/hooks/scripts/util.sh
+unset_secrets () {
```
Comment (dliappis) [inline .buildkite/hooks/pre-exit:5]: In the future we could think about complete deleting this hook. All agents (k8s or vms) are ephemeral.

File: .buildkite/x-pack/pipeline.xpack.metricbeat.yml [hunk 2]
```diff
@@ -81,6 +81,10 @@ steps:
         key: "mandatory-int-test"
         command: |
           set -euo pipefail
+          # docker-compose problem debugging
+          which docker-compose
+          docker-compose --version
+          pwd
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.metricbeat.yml:87]: is this still needed?

**Comment by oakrizan** [comment]
@dliappis PTAL 🙏

**Review by dliappis — APPROVED** [review]
LGTM (I understand that the k8s failures are tracked in https://github.com/elastic/ingest-dev/issues/3549) but given that we are changing k8s related scripts here, I'd feel safer to merge this once we have green CI.

File: .buildkite/hooks/pre-exit (hunk 1 above)
Comment (oakrizan) [reply .buildkite/hooks/pre-exit:5]: I will create a separate PR to delete it.

File: .buildkite/x-pack/pipeline.xpack.metricbeat.yml (hunk 2 above)
Comment (oakrizan) [reply .buildkite/x-pack/pipeline.xpack.metricbeat.yml:87]: Sorry, accidentally added it. Will remove.

**Comment by oakrizan** [comment]
> LGTM (I understand that the k8s failures are tracked in [elastic/ingest-dev#3549](https://github.com/elastic/ingest-dev/issues/3549)) but given that we are changing k8s related scripts here, I'd feel safer to merge this once we have green CI.

Yep, I'm working on the issue with k8s version update. But taking into account that `v1.21.4` [step was successful](https://buildkite.com/elastic/deploy-k8s/builds/5373#01909bdf-2d25-4966-884c-74d63c0b082f), I'd say those changes were validated. WDYT @dliappis?

**Comment by oakrizan** [comment]
Agreed to force-merge: https://elastic.slack.com/archives/C06RF2SAWMS/p1720694503396459

**Pull request merged**